androidApp/src/main/java/com/mbta/tid/mbta_app/android/stopDetails/StopDetailsViewModel.kt

```diff
@@ -1,5 +1,217 @@
 package com.mbta.tid.mbta_app.android.stopDetails
 
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.compose.LifecycleResumeEffect
+import com.mbta.tid.mbta_app.android.state.ScheduleFetcher
+import com.mbta.tid.mbta_app.android.state.StopPredictionsFetcher
+import com.mbta.tid.mbta_app.android.util.timer
+import com.mbta.tid.mbta_app.model.StopDetailsDepartures
+import com.mbta.tid.mbta_app.model.response.AlertsStreamDataResponse
+import com.mbta.tid.mbta_app.model.response.GlobalResponse
+import com.mbta.tid.mbta_app.model.response.PredictionsByStopJoinResponse
+import com.mbta.tid.mbta_app.model.response.PredictionsByStopMessageResponse
+import com.mbta.tid.mbta_app.model.stopDetailsPage.StopData
+import com.mbta.tid.mbta_app.repositories.IErrorBannerStateRepository
+import com.mbta.tid.mbta_app.repositories.IPredictionsRepository
+import com.mbta.tid.mbta_app.repositories.ISchedulesRepository
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.flow.updateAndGet
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.koin.androidx.compose.koinViewModel
 
-class StopDetailsViewModel() : ViewModel() {}
+class StopDetailsViewModel(
+    schedulesRepository: ISchedulesRepository,
+    private val predictionsRepository: IPredictionsRepository,
+    private val errorBannerRepository: IErrorBannerStateRepository,
+) : ViewModel() {
+    private val stopScheduleFetcher = ScheduleFetcher(schedulesRepository, errorBannerRepository)
+
+    private val _stopData = MutableStateFlow<StopData?>(null)
+    val stopData: StateFlow<StopData?> = _stopData
+
+    val _stopDepartures = MutableStateFlow<StopDetailsDepartures?>(null)
+    val stopDepartures: StateFlow<StopDetailsDepartures?> = _stopDepartures
```

Comment:
Departures aren't stored in the iOS view model, but I think it makes things a bit easier to work with on the android side. The departures need to get passed from NearbyTransit => StopDetails so that they aren't recalculated & temporarily null when a stop filter changes, so may as well pass it on the VM.